tests/conftest.py

```diff
@@ -48,3 +49,28 @@ def undo():
             monkeypatch.setattr('click.echo', old_echo)
 
     return Result
+
+
+@pytest.fixture(scope='session')
+def sleep_time(tmpdir_factory):
+    """
+    Returns how long we need to sleep for the filesystem's mtime precision to
+    pick up differences.
+
+    This keeps test fast on systems with high precisions, but makes them pass
+    on those that don't.
+    """
+    tmpfile = tmpdir_factory.mktemp('sleep').join('touch_me')
+
+    def touch_and_mtime():
+        tmpfile.open('w').close()
+        stat = os.stat(str(tmpfile))
+        return getattr(stat, 'st_mtime_ns', stat.st_mtime)
+
+    first = touch_and_mtime()
+    sleep(0.1)
+    second = touch_and_mtime()
+
+    if first < second:
+        return 0.01
```

Comment:
You're right.

We could also get greedy and loop `[0.001, 0.01, 0.1, 1]` and return the first value for which `first != second`:

```
for i in [0.001, 0.01, 0.1, 1]:
    first = touch_and_mtime()
    sleep(i)
    second = touch_and_mtime()
    if first != second:
        return i
```